### PR TITLE
Replace dynamic on_trait_change with observe part 2

### PR DESCRIPTION
**Pull request by aaronayres35**
This PR continues replacing the dynamic uses of on_trait_change with observe equivalents on top of #590

### CODE REVIEW

File: chaco/tools/range_selection.py [hunk 1]
```diff
@@ -679,15 +679,15 @@ def _determine_axis(self):
...
         return
 
     def _axis_changed(self, old, new):
         if old is not None:
-            self.plot.on_trait_change(self.__mapper_changed,
+            self.plot.observe(self.__mapper_changed,
                                       old + "_mapper", remove=True)
         if new is not None:
-            self.plot.on_trait_change(self.__mapper_changed,
-                                      old + "_mapper", remove=True)
```
Comment (aaronayres35) [inline chaco/tools/range_selection.py:692]: This previous code seems like it was a copy paste error...
Butt the updated code is still not right ad we have a case in the test suite where old is non None, but its listener never was hooked up initially

File: chaco/scatter_inspector_overlay.py [hunk 2]
```diff
@@ -41,7 +42,7 @@ class ScatterInspectorOverlay(AbstractOverlay):
     # using a series of trait change handlers (defined at the end of the
     # class)
```
Comment (rahulporuri) [inline chaco/scatter_inspector_overlay.py:43]: This is tangential but i want to dig a little and find the source of this comment - it looks like there was going to be some fancy traits feature which would remove unnecessary code.

File: chaco/scatter_inspector_overlay.py [hunk 3]
```diff
@@ -41,7 +42,7 @@ class ScatterInspectorOverlay(AbstractOverlay):
...
     #@on_trait_change('component.index.metadata_changed,component.value.metadata_changed')
-    def metadata_changed(self, object, name, old, new):
+    def metadata_updated(self, event):
```
Comment (rahulporuri) [inline chaco/scatter_inspector_overlay.py:45]: public API - just being safe

```suggestion
    def metadata_updated(self, event=None):
```

**Review by rahulporuri — APPROVED** [review]
LGTM with a couple of small comments.

There are a bunch of indentation errors that i'm ignoring because we'll run black on chaco at somepoint this month, which should fix all of those issues.

**Pull request merged**